kotlin-plugin/src/main/kotlin/me/shika/generation/ObjectSerializationIRGeneration.kt

```diff
@@ -17,7 +17,7 @@ import org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid
 import org.jetbrains.kotlin.ir.visitors.acceptChildrenVoid
 import org.jetbrains.kotlin.ir.visitors.acceptVoid
 
-private object ObjectSerializationOrigin : IrDeclarationOriginImpl("object-serialization-fix", isSynthetic = true)
+private val objectSerializationOrigin = IrDeclarationOriginImpl("object-serialization-fix", isSynthetic = true)
```

Comment:
THIS IS THE BREAKING CHANGE in Kotlin 2.0
* Before `IrDeclarationOriginImpl` used be a `abstract class`.
* Now, `IrDeclarationOriginImpl` is a `final class`. So we can't just extend it like we used to.

This change is backward-incompatible with Kotlin <2, so we should consider releasing a major version and documenting that that the new version only supports Kotlin 2+.